store/src/java/com/zimbra/cs/mailbox/DistributedMailboxCacheManager.java

```diff
@@ -0,0 +1,220 @@
+        }
+    }
+
+    @Override
+    public int getMailboxCount(){
+        Integer mailboxCount = null;
+        DbPool.DbConnection conn = null;
+        try {
+            try {
+                conn = DbPool.getConnection();
+                mailboxCount =  DbMailbox.getMailboxCount(conn);
+            } catch (ServiceException e) {
+                ZimbraLog.mailbox.error("getting mailbox count: ", e);
+                throw new CacheManagerException("getting mailbox count: ", e.getCause());
+            }
+        }finally {
+            DbPool.quietClose(conn);
+        }
+        return mailboxCount;
+    }
+
+    @Override
+    public Set<String> getAccountIds(){
+        return listMailboxes().keySet();
+    }
+
+    @Override
+    public Set<Map.Entry<String, Integer>> getMailboxIdsByAccountId() {
+        return listMailboxes().entrySet();
+    }
+
+    public Collection<Integer> getMailboxIds(){
+        return listMailboxes().values();
+    }
+
+
+    @Override
+    public Integer getMailboxKey(String accountId){
+        Integer mailboxKey = null;
+        DbPool.DbConnection conn = null;
+        try {
+            try {
+                conn = DbPool.getConnection();
+                mailboxKey =  DbMailbox.getMailboxKey(conn, accountId);
+            } catch (ServiceException e) {
+                ZimbraLog.mailbox.error("fetching mailbox key: ", e);
+                throw new CacheManagerException("fetching mailbox key: ", e.getCause());
+            }
+        }finally {
+            DbPool.quietClose(conn);
+        }
+        return mailboxKey;
+    }
+
+    @Override
+    public void removeMailboxId(String accountId) {
+        //noop
+    }
+
+    @Override
+    public Object retrieveFromCache(int mailboxId, boolean trackGC, MailboxManager mailboxManager) throws MailServiceException {
+        return getMailbox(mailboxId);
+
+    }
+
+    @Override
+    public void cacheMailbox(int mailboxId, MailboxMaintenance maint) {
+        //noop
+    }
+
+    @Override
+    public Object getMailbox(int mailboxId) {
+        Mailbox.MailboxData data;
+        DbPool.DbConnection conn = null;
+        try {
+            conn = DbPool.getConnection();
+            data = DbMailbox.getMailboxStats(conn, mailboxId);
+            if (data == null) {
+                throw MailServiceException.NO_SUCH_MBOX(mailboxId);
+            }
+        } catch (MailServiceException e) {
+            ZimbraLog.mailbox.error("fetching mailbox: ", e);
+            throw new CacheManagerException("fetching mailbox: ", e.getCause());
+        } catch (ServiceException e) {
+            ZimbraLog.mailbox.error("fetching mailbox: ", e);
+            throw new CacheManagerException("fetching mailbox: ", e.getCause());
+        } finally {
+            conn.closeQuietly();
+        }
+        return new Mailbox(data);
+    }
+
+    @Override
+    public Mailbox cacheMailbox(Mailbox mailbox, MailboxManager mailboxManager) {
```

Comment:
`MailboxManager mailboxManager` does not appear to be used here.

Reply:
removed, its not being used